include/miopen/miopen.h

```diff
@@ -2428,6 +2440,55 @@ MIOPEN_EXPORT miopenStatus_t miopenDestroyLRNDescriptor(miopenLRNDescriptor_t lr
 /** @} */
 // CLOSEOUT LRN DOXYGEN GROUP
 
+#ifdef MIOPEN_BETA_API
```

Comment:
How this macro can be defined? It must be defined in this PR in order to feed the new code into cppcheck and clang-tidy and in order to test the new primitive.

Reply:
I was asked by @TejashShah that the new API should be guarded by MIOPEN_BETA_API macro. So I thought there was a policy for MIOPEN_BETA_API internally in CI/CD, but is there anything else I need to do about this?

Reply:
The idea is that the customer who would like to use this feature would define the macro when building their app. This informs the customer that this is a beta feature and might be revised in the near future. Furthermore, this allows MIOpen to revise the API semantics if needed since the customer is aware that they are using a beta feature hence change of API is acceptable.

Reply:
I agree we should have more documentation, which can follow in a subsequent PR however I dont want us to introduce a new macro for each new API we introduce.

Reply:
@JehandadKhan Well, actually we can get rid of both these "enable" and "api version" macros after experimental feature becomes the regular one (at the small cost of breaking experimental apps -- and that will be trivial to fix).

Reply:
@atamazov Doesn't a single MIOPEN_BETA_API flag, as proposed by @JehandadKhan and implemented by @seungmanhan,  serve the purpose to indicate it is experimental? We could use the same flag for additional kernels expected from @seungmanhan. 

@seungmanhan could address your comments pertaining to removing conditionals based on this flag for unit tests and driver code as we would want to test the code even it is in a beta form.